Data Quality Checks.Rmd

```diff
@@ -13,90 +11,101 @@ knitr::opts_chunk$set(echo = TRUE)
 
 ```{r, include=FALSE}
 
-##Libraries & set up (maybe don't need to load all of these if data checks are incorporated to main indicator production script?)**
-
-library(dplyr) # for data manipulation
-library(ggplot2) # for plotting
-library(tidyr) # for data manipulation
-library(readr) # writing csv's
-library(readxl) #for reading excel
-library(reshape2) # for dcasting/reshape
-library(flextable) # for output tables
-library(rmarkdown)
-
-# Varies filepaths depending on if using server or not and what organisation uses it.
-if (exists("organisation") == TRUE) { #Health Scotland
-  if (organisation == "HS") { 
-    data_folder <- "X:/ScotPHO Profiles/Data/" 
-    lookups <- "X:/ScotPHO Profiles/Data/Lookups/"
-  }
-} else  { #ISD, first server then desktop
-  if (sessionInfo()$platform %in% c("x86_64-redhat-linux-gnu (64-bit)", "x86_64-pc-linux-gnu (64-bit)")) {
-    data_folder <- "/PHI_conf/ScotPHO/Profiles/Data/"
-    lookups <- "/PHI_conf/ScotPHO/Profiles/Data/Lookups/" 
-  } else {
-    data_folder <- "//stats/ScotPHO/Profiles/Data/"
-    lookups <- "//stats/ScotPHO/Profiles/Data/Lookups/" 
-  }
-}
-
-#Set parameters
-filename <- params$data #setting parameter to file name makes subsequent code neater? 
-
-#Pre-defined geographies that will be used in 'Check 3' comparing old & new file - maybe wants to use parameters?
-check_codes =c("S00000001","S08000015","S12000010","S02001236") 
+#read in geography code names dictionary
+code_dictionary <- readRDS(paste0(data_folder,"Lookups/Geography/codedictionary.rds")) 
 
 #Read in indicator file being checked
 data_indicator <- readRDS(paste0(data_folder, "Data to be checked/", filename, "_shiny.rds")) 
 
+#Add geography code names
+data_indicator <- left_join(x=data_indicator, y=code_dictionary, by="code")
+
+#Test which optional geographies are present in dataset
+board <- ifelse(as.numeric(count(data_indicator %>% filter(substr(code,1,3) == "S08"))) > 0,TRUE, FALSE)
```

Comment:
These tests could be simplified using any (all is also an useful function):
any(substr(data_indicator$code,1,3) == "S08")

Reply:
Done